pkg/helm/repo.go

```diff
@@ -79,7 +81,18 @@ func RepoAdd(ctx context.Context, logger logr.Logger, name, url string) (err err
 		return errors.Errorf("repository name (%s) contains '/', please specify a different name without '/'", entry.Name)
 	}
 
-	r, err := repo.NewChartRepository(&entry, getter.All(settings))
+	r, err := repo.NewChartRepository(&entry, getter.Providers{
```

Comment:
Great!
What about make it a function and also add pluginDownloaders?
```go
pluginDownloaders, _ := collectPlugins(settings)
result = append(result, pluginDownloaders...)
```
I guess we can support helm plugin, just let user rebuild core image or use init containers to add their plugin binary to the specified location